wip-trampoline.md

```diff
@@ -187,12 +187,97 @@ themselves.
 
 This grows the maximum route length to `9 * 8 = 72` total hops.
 
+## Network Pruning
+
+TODO: this section is not in spec format at all. It's a draft of the overall
+ideas to make sure we agree on the path forward before starting a spec effort.
+
+Nodes running on constrained devices (phones, IoT, etc) should only keep track
+of nearby channels (with an `N`-radius heuristic for example). These nearby
+channels will be used to build a v0 onion route to a first trampoline node.
+
+Such nodes can also choose to ignore channels with a capacity lower than
+`min_chan_capacity`. `N` and `min_chan_capacity` are configured by the node and
+not advertised to the network.
+
+Such nodes would simply ignore `channel_announcement`s that don't meet those
+requirements.
+
+Such nodes would also need to store information about nodes that are outside of
+their network view to use as trampolines.
+
+We introduce a new `node_update` message with the following structure:
+
+1. type: 260 (`node_update`)
+2. data:
+   * [`64`:`signature`]
+   * [`32`:`chain_hash`]
+   * [`33`:`node_id`]
+   * [`4`:`timestamp`]
+   * [`2`:`cltv_expiry_delta`]
+   * [`8`:`htlc_minimum_msat`]
+   * [`8`:`htlc_maximum_msat`]
+   * [`4`:`fee_base_msat`]
+   * [`4`:`fee_proportional_millionths`]
```

Comment:
Unless this message is also gated by the existence of a channel with the node as an endpoint it becomes a really cheap DoS attack vector. It's also hard for a node with a partial network view to even verify that there is a channel matching this node.

Reply:
Nodes will apply the same filters for this message as what they do for `channel_update`s.
A node producing the `node_update` can only send it to peers with whom it has a channel and can't send `node_update`s more than once every X seconds.
Most nodes that support the `node_update` message will be routing/trampoline nodes and will have the full network view, so they can drop messages from nodes that don't have any channel.

You have a very good point though for the constrained nodes that have only a view of their neighbourhood (let's call one of those Alice). If one of Alice's neighbours is malicious and forwards `node_update`s for nodes that don't have any channel, Alice can't verify that and might include that node in her trampoline routes which is an issue. However, Alice will regularly change her list of trampoline nodes based on randomness and reliability: payment failure will trigger the eviction of the malicious node.

That needs to be detailed more thoroughly though.